src/gc/gc.d

```diff
@@ -3260,3 +3264,32 @@ else
         return p;
     }
 }
+
+debug (MEMSTOMP)
+unittest
+{
+	import core.memory;
+	auto p = cast(uint*)GC.malloc(uint.sizeof*3);
+	assert(*p == 0xF0F0F0F0);
+	p[2] = 0; // First two will be used for free list
+	GC.free(p);
+	assert(p[2] == 0xF2F2F2F2);
+}
+
+debug (SENTINEL)
+unittest
+{
+	import core.memory;
+	auto p = cast(ubyte*)GC.malloc(1);
+	assert(p[-1] == 0xF4);
+	assert(p[ 1] == 0xF5);
+
+	p[1] = 0;
+	bool thrown;
+	try
+		GC.free(p);
+	catch (Error e)
```

Comment:
You could used version(unittest) to change the behaviour of the invariant, so it could just count errors while executing this unittest. Not sure how important that is here, though. Maybe just add a comment why the test isn't executed.